storehaus-mysql/src/main/scala/com/twitter/storehaus/mysql/MySqlStore.scala

```diff
@@ -0,0 +1,272 @@
+    val mysqlResult =
+      toTwitterFuture(String2MySqlValueInjection.invert(k)).flatMap { str =>
+        selectStmt.select(str.getBytes) { row => row(vCol).map(MySqlValue(_)) }
+      }
+
+    mysqlResult.map(_.headOption.flatten)
+  }
+
+  override def multiGet[K1 <: MySqlValue](ks: Set[K1]): Map[K1, Future[Option[MySqlValue]]] = {
+    if (ks.isEmpty) {
+      Map.empty
+    } else {
+      // build preparedstatement based on keyset size
+      val selectSql = MULTI_SELECT_SQL_PREFIX + buildPlaceholders("?", ks.size, ",", "(", ")")
+
+      val params = toTwitterFuture(
+        ks.map(String2MySqlValueInjection.invert(_).map(s => Parameter.wrap(s.getBytes))).toSeq)
+      val mysqlResult = params.flatMap { ps =>
+        client.prepare(selectSql).select(ps: _*) { row =>
+          (row(kCol).map(MySqlValue(_)), row(vCol).map(MySqlValue(_)))
+        }
+      }
+      FutureOps.liftValues(
+        ks,
+        mysqlResult.map(rows => rows.iterator.collect { case (Some(k), v) => (k, v) }.toMap),
+        (k: K1) => Future.None
+      )
+    }
+  }
+
+  protected def set(k: MySqlValue, v: MySqlValue) = doSet(k, v)
+
+  override def put(kv: (MySqlValue, Option[MySqlValue])): Future[Unit] = {
+    kv match {
+      case (key, Some(value)) => doSet(key, value).unit
+      case (key, None) => doDelete(key).unit
+    }
+  }
+
+  override def multiPut[K1 <: MySqlValue](
+      kvs: Map[K1, Option[MySqlValue]]): Map[K1, Future[Unit]] = {
+    // batched version of put. the batch is split into insert, update, and delete statements.
+    // reduce your batch size if you are hitting mysql packet limit:
+    // http://dev.mysql.com/doc/refman/5.1/en/packet-too-large.html
+    val putResult = startTransaction.flatMap { t =>
+      FutureOps.mapCollect(multiGet(kvs.keySet)).flatMap { result =>
+        val existingKeys = result.filter(_._2.isDefined).keySet
+        val newKeys = result.filter(_._2.isEmpty).keySet
+
+        // handle inserts for new keys
+        val insertF =
+          if (newKeys.isEmpty) {
+            Future.Unit
+          } else {
+            // do not include None values in insert query
+            val insertKvs = newKeys.iterator
+              .collect { case k if kvs.contains(k) && kvs(k).isDefined => k -> kvs(k) }
+              .toMap.mapValues(_.get)
+            if (insertKvs.isEmpty) Future.Unit
+            else executeMultiInsert(insertKvs).unit
+          }
+
+        // handle update and/or delete for existing keys
+        val existingKvs = existingKeys.map(k => k -> kvs.getOrElse(k, None))
+
+        // do not include None values in update query
+        val updateKvs = existingKvs.iterator.collect { case (k, Some(v)) => (k, v) }.toMap
+        lazy val updateF =
+          if (updateKvs.isEmpty) Future.Unit
+          else executeMultiUpdate(updateKvs).unit
+
+        // deletes
+        val deleteKeys = existingKvs.iterator.collect { case (k, None) => k }.toSeq
```

Comment:
the `.toSeq` was missing as a result the iterator was consumed by `deleteKeys.size` when building the sql request and `deleteParams` was empty.